functions/send_action_list.ts

```diff
@@ -0,0 +1,59 @@
+import { DefineFunction, Schema, SlackFunction } from "deno-slack-sdk/mod.ts";
+import { ActionItemInfo } from "../types/action_item_info.ts";
+
+export const SendActionFunction = DefineFunction({
+  callback_id: "send_action",
+  title: "Send Action Item in a message",
+  description: "Send a message with the specified action items.",
+  source_file: "functions/send_action_list.ts",
+  input_parameters: {
+    properties: {
+      action_items: {
+        type: Schema.types.array,
+        items: {
+          type: ActionItemInfo,
+        },
+      },
+      channel: {
+        type: Schema.slack.types.channel_id,
+      },
+      thread_ts: {
+        type: Schema.slack.types.message_ts,
+        description: "optionally used to send message as a response",
+      },
+    },
+    required: ["action_items", "channel"],
+  },
+});
+
+export default SlackFunction(
+  SendActionFunction,
+  async ({ inputs, client }) => {
+    const { action_items, channel, thread_ts } = inputs;
+
+    let message = "";
+    if (action_items.length) {
+      message += "*User Action List:*\n";
+      message += action_items.map((item) =>
+        actionItemToMarkdownBullet(item.name, item.end_date)
+      ).join("\n");
+    } else {
+      message += "This user does not have any action items.";
+    }
+
+    if (channel && message) {
+      await client.chat.postMessage({
```

Comment:
Since the text action list doesn't mention the name of the user or anything, this would be kind of silly to see in a shared channel and mean nothing to everyone else. Maybe we can use an ephemeral message instead? I think we would need to also pass in a user id as well if we go that direction. https://api.slack.com/methods/chat.postEphemeral/code